packages/engine/Specs/Scene/ScreenSpaceCameraControllerSpec.js

```diff
@@ -1566,6 +1566,79 @@ describe("Scene/ScreenSpaceCameraController", function () {
     expect(camera.direction).not.toEqual(direction);
   });
 
+  it("maximum tilt angle is 0 in 3D", function () {
+    setUp3D();
+    const originalCameraPosition = Cartesian3.clone(camera.position);
+    const originalCameraDirection = Cartesian3.clone(camera.direction);
+    controller.maximumTiltAngle = 0;
+    const startPosition = new Cartesian2(
+      canvas.clientWidth / 2,
+      canvas.clientHeight / 2
+    );
+    const endPosition = new Cartesian2(
+      canvas.clientWidth / 2,
+      canvas.clientHeight / 4
+    );
+
+    moveMouse(MouseButtons.MIDDLE, startPosition, endPosition);
+    updateController();
+
+    expect(camera.position).toEqualEpsilon(
+      originalCameraPosition,
+      CesiumMath.EPSILON8
+    );
+    expect(camera.position).not.toEqualEpsilon(
+      originalCameraDirection,
+      CesiumMath.EPSILON8
+    );
+
+    const dotProduct = Cartesian3.dot(
+      camera.direction,
+      originalCameraDirection
+    );
+    const acos = Math.acos(dotProduct);
+    const angle = (acos * 180) / Math.PI;
+    expect(angle).toEqual(0);
+  });
+
+  it("maximum tilt angle is 45 degrees in 3D", function () {
+    setUp3D();
+    const originalCameraPosition = Cartesian3.clone(camera.position);
+    const originalCameraDirection = Cartesian3.clone(camera.direction);
+    controller.maximumTiltAngle = (45 * Math.PI) / 180;
+    const startPosition = new Cartesian2(
+      canvas.clientWidth / 2,
+      canvas.clientHeight / 2
+    );
+    const endPosition = new Cartesian2(
+      canvas.clientWidth / 2,
+      canvas.clientHeight / 4
+    );
+
+    moveMouse(MouseButtons.MIDDLE, startPosition, endPosition);
```

Comment:
why repeat this mouse move? If it is valid can you add a comment to explain?

Reply:
Certainly.
The reason why I repeated the mouse move was because I could not find a way to get the tilting angle to 45 degrees (0.25π radians) in just one move. I tried to set the endPosition to different values but could not get the final tilt angle high enough to the limit I wanted to test. Maybe I am missing something, and there is a better way?